addons/product/models/product_template.py

```diff
@@ -502,6 +543,226 @@ def has_dynamic_attributes(self):
+        context = {
+            'quantity': add_qty,
+            'pricelist': pricelist.id if pricelist else None,
+            'current_attributes_price_extra': [v.price_extra or 0.0 for v in product_template_attribute_values]
+        }
+
+        product_template = self.env['product.template'].with_context(context).browse(product_template_id)
+
+        Product = self.env['product.product'].with_context(context)
+        if product_id and not combination:
+            product = Product.browse(product_id)
+        else:
+            products = Product.search([
+                ('product_tmpl_id', '=', product_template_id)
+            ])
+            product = products.filtered(
+                lambda product: all(product_attribute_value in product.product_template_attribute_value_ids
+                    for product_attribute_value in filtered_product_template_attribute_values)
+            )
+
+        product_id = None
+        list_price = product_template.price_compute('list_price')[product_template.id]
+        price = product_template.price or list_price
+        if(product):
+            product = product.with_context(
+                no_variant_attributes_price_extra=[product_attribute_value.price_extra or 0.0
+                    for product_attribute_value in product_template_attribute_values.filtered(
+                        lambda product_attribute_value: product_attribute_value.attribute_id.create_variant == 'no_variant'
+                    )
+                ]
+            )
+            product_id = product.id
+            list_price = product.price_compute('list_price')[product.id]
+            price = product.price or list_price
+
+        display_name = [product_template.name]
+        if filtered_product_template_attribute_values:
+            display_name.append(' (')
+            display_name.append(', '.join(filtered_product_template_attribute_values.mapped('name')))
+            display_name.append(')')
+
+        if pricelist and pricelist.currency_id != product_template.currency_id:
+            list_price = product_template.currency_id._convert(
+                list_price, pricelist.currency_id,
+                self.env.user.company_id, fields.Date.today()
+            )
+
+        return {
+            'product_id': product_id,
+            'product_template_id': product_template.id,
+            'display_name': ''.join(display_name),
+            'price': price,
+            'list_price': list_price
+        }
+
+    @api.multi
+    def is_combination_possible(self, combination, reference_product=None):
+        """
+            :param combination: recordset of product.template.attribute.value
+            :param reference_product: `product.product` from which 'self' is an
+                optional or accessory product
+            :return: boolean: True if the combination is possible
+        """
+        self.ensure_one()
+        combinationData = self._get_attribute_exclusions(reference_product=None)
```

Comment:
@seb-odoo 

Why do you explicitly pass "reference_product=None" instead of passing the one you received as argument of the method?
Even though I don't think you ever call this method with a reference_product as parameter, it still looks wrong to me.

Reply:
This is fixed now, using the correct parameter.